### PR TITLE
feat(bedrock): add prompt caching middleware

### LINKED ISSUE #9831: Feature Request: Bedrock-Compatible PromptCaching Middleware
### Checked other resources

- [x] This is a bug, not a usage question. For questions, please use the LangChain Forum (https://forum.langchain.com/).
- [x] I added a very descriptive title to this issue.
- [x] I searched the LangChain.js documentation with the integrated search.
- [x] I used the GitHub search to find a similar question and didn't find it.
- [x] I am sure that this is a bug in LangChain.js rather than my code.
- [x] The bug is not resolved by updating to the latest stable version of LangChain (or the specific integration package).

### Example Code

Sample usage:
```typescript
import { ChatBedrockConverse } from '@langchain/aws';
import { createAgent, bedrockPromptCachingMiddleware } from 'langchain';

// Example usage:
const model = new ChatBedrockConverse({
  model: 'anthropic.claude-3-5-sonnet-20241022-v2:0',
  region: 'us-east-1'
});

const agent = createAgent({
  model,
  middleware: [
    bedrockPromptCachingMiddleware({
      enableCaching: true,
      minMessagesToCache: 10
    })
  ],
});

// Invoke the agent
const result = await agent.invoke({
  messages: [{ role: 'user', content: 'Hello!' }]
});
```

### Error Message and Stack Trace (if applicable)

_No response_

### Description

**Summary:**
Requesting support for Bedrock models in LangChain's promptCaching middleware, similar to the implementation for Anthropic found in [`libs/langchain/src/agents/middleware/provider/anthropic/promptCaching.ts`](https://github.com/langchain-ai/langchainjs/blob/35df8fb592c69d482520ee3ae1a60b05dd48bbb0/libs/langchain/src/agents/middleware/provider/anthropic/promptCaching.ts).

**Details:**
- Ensure cache control logic and API structure are compatible with Bedrock endpoints and formats.
- Matching feature set to Anthropic:
    - Configurable minMessagesToCache
    - Runtime and middleware options for enableCaching
    - Behavior control for unsupported models
- Consider Bedrock-specific API behaviors and cache headers. 

**Benefits:**
- Cost and latency reduction for repetitive prompts, long system messages, or extensive conversation histories on Bedrock.
- Enables prompt caching best practices across more providers.

**References:**
- Existing Anthropic middleware: [`promptCaching.ts`](https://github.com/langchain-ai/langchainjs/blob/35df8fb592c69d482520ee3ae1a60b05dd48bbb0/libs/langchain/src/agents/middleware/provider/anthropic/promptCaching.ts)
- [Anthropic prompt caching docs](https://docs.anthropic.com/en/docs/build-with-claude/prompt-caching)

**Suggested implementation notes:**
- Structure middleware API and error handling similar to Anthropic version.
- Document differences and caveats between Anthropic and Bedrock behavior.
- Add examples and usage guidance for Bedrock prompt caching in documentation.

### System Info

N/A

### PR DESCRIPTION
**Pull request by midikaif**
## Description
This PR adds Prompt Caching support for AWS Bedrock, similar to the existing Anthropic implementation.

## Changes
- Added `bedrockPromptCachingMiddleware` in `libs/langchain/src/agents/middleware/provider/bedrock/promptCaching.ts`
- Added comprehensive unit tests handling edge cases.

## Testing
I have added a test suite that verifies:
- [x] Middleware correctly adds the `cache_control` tag to the last message.
- [x] Respects `minMessagesToCache` threshold.
- [x] Handles `enableCaching: false` correctly.
- [x] Safely handles `undefined` or empty inputs.
- [x] Skips non-text content (e.g. images) to prevent corruption.
- [x] Skips Tool/Function messages.

Tests ran successfully locally using `vitest`.

Fixes #9831

### CODE REVIEW

File: libs/providers/langchain-cloudflare/package.json [hunk 1]
```diff
@@ -58,14 +58,14 @@
   "types": "./dist/index.d.cts",
   "exports": {
     ".": {
-      "input": "./src/index.ts",
+      "input": "./src\\index.ts",
       "require": {
-        "types": "./dist/index.d.cts",
-        "default": "./dist/index.cjs"
+        "types": "./dist\\index.d.cts",
+        "default": "./dist\\index.cjs"
```
Comment (christian-bromann) [inline libs/providers/langchain-cloudflare/package.json]: It seems like there are a lot of these unrelated changes, can we revert them?

File: libs/langchain-core/package.json [hunk 2]
```diff
@@ -76,674 +76,674 @@
...
-        "types": "./dist/index.d.cts",
-        "default": "./dist/index.cjs"
+        "types": "./dist\\index.d.cts",
+        "default": "./dist\\index.cjs"
       },
       "import": {
-        "types": "./dist/index.d.ts",
-        "default": "./dist/index.js"
+        "types": "./dist\\index.d.ts",
+        "default": "./dist\\index.js"
```
Comment (christian-bromann) [inline libs/langchain-core/package.json]: There are still these changes

**Comment by midikaif** [comment]
@christian-bromann Thanks for the patience. I've cleaned up the commit history and verified that all unrelated package.json changes are removed. The PR now only contains the Bedrock middleware and relevant tests.

**Comment by christian-bromann** [comment]
@midikaif two requests:
- mind checking the CI issues
- mind raising a docs PR for [this file](https://github.com/langchain-ai/docs/blob/main/src/oss/langchain/middleware/built-in.mdx)?

File: libs/langchain/src/agents/middleware/provider/bedrock/promptCaching.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+export function bedrockPromptCachingMiddleware(
+  config: BedrockPromptCachingConfig = {}
+) {
+  const { enableCaching = true, minMessagesToCache = 2 } = config;
+
+  return <RunInput, RunOutput>(
+    runnable: Runnable<RunInput, RunOutput>
+  ): Runnable<RunInput, RunOutput> => {
```
Comment (christian-bromann) [inline libs/langchain/src/agents/middleware/provider/bedrock/promptCaching.ts:16]: We should return a `AgentMiddleware` here.

File: libs/langchain/src/agents/middleware/provider/bedrock/tests/promptCaching.test.ts [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+    ];
+
+    await runnable.invoke({ messages });
+
+    const lastMsg = messages[3];
+    
+    expect(lastMsg.additional_kwargs.cache_control).toEqual({
+      type: "ephemeral",
+    });
+  });
```
Comment (christian-bromann) [inline libs/langchain/src/agents/middleware/provider/bedrock/tests/promptCaching.test.ts:83]: These tests should use `createAgent`

**Review by christian-bromann — COMMENTED** [review]
It seems the implementation doesn't implement an actual middleware. Instead it returns a Runnable. I think we should look at this from the perspective of a createAgent user and expose it as middleware.
